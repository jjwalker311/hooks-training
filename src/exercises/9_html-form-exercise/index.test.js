import React from 'react'
// import user from '@testing-library/user-event'
// import {
//   render, screen,
// } from '@testing-library/react'

import Exercise9 from '.'
import { shallow } from 'enzyme'

// const USERNAME = 'some-user-name'
// const ERROR_MESSAGE = 'some-error-message'

const mockUpdateUserName = jest.fn(() => Promise.resolve())

const wrapper = shallow(<Exercise9 updateUsername={mockUpdateUserName} />)

// const handleSubmit = wrapper.find('form').prop('onSubmit')

// function isInError() {
//   return !!wrapper.find('span.error').text()
// }

// describe('Suite of tests on Exercise 9', () => {
//   it('should render basic form items', () => {
//     expect(snapshot(wrapper)).toMatchSnapshot()
//   })

//   it('calls updateUsername with the new username on form submit', async () => {
//   })

//   it('displays error message when no username is entered', async () => {
//   })
// })
