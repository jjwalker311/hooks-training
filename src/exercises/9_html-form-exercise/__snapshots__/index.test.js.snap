// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suite of tests on Exercise 9 should render basic form items 1`] = `
<form
  onSubmit={[Function]}
>
  <br />
  <br />
  <label
    htmlFor="username"
  >
    Username
    <input
      id="username"
    />
  </label>
  <button
    name="submit"
    type="submit"
  >
    Submit
  </button>
  <span
    className="status"
  />
  <span
    className="error"
  />
</form>
`;
